pkg/kapp/clusterapply/converged_resource.go

```diff
@@ -41,6 +70,35 @@ func (c ConvergedResource) IsDoneApplying() (ctlresm.DoneApplyState, []string, e
 		return ctlresm.DoneApplyState{Done: true}, descMsgs, err
 	}
 
+	// Custom wait rule
+	wr := c.waitingRule
```

Comment:
instead of adding all this logic here let's define a new object in https://github.com/k14s/kapp/tree/master/pkg/kapp/resourcesmisc. once that's done we should be able to add that to list of resource waiters here: https://github.com/k14s/kapp/blob/master/pkg/kapp/clusterapply/converged_resource_factory.go#L25 (probably makes sense to add that to the top to prefer custom provided waiting over included ones).

Reply:
I had initially considered that, but then it is exclusive to only the custom rules set.
https://github.com/k14s/kapp/blob/a5d09876ea628695323920e7a86bddab9a7fb912/pkg/kapp/clusterapply/converged_resource.go#L156
I'm fine with this.